openfe/setup/atom_mapping/perses_mapper.py

```diff
@@ -50,24 +51,31 @@ def __init__(self, allow_ring_breaking: bool = True,
         self.coordinate_tolerance = coordinate_tolerance
 
     def _mappings_generator(self, molA, molB):
+        # Construct Perses Mapper
         _atom_mapper = AtomMapper(
             use_positions=self.use_positions,
             coordinate_tolerance=self.coordinate_tolerance,
             allow_ring_breaking=self.allow_ring_breaking)
 
-        # Type of mapping
+        # Try generating a mapping
         try:
             _atom_mappings = _atom_mapper.get_all_mappings(
                     old_mol=molA.to_openff(), new_mol=molB.to_openff())
         except InvalidMappingException:
             _atom_mappings = []
 
+        # Catch empty mappings here
+        if(_atom_mappings is None):
+            _atom_mappings = []
```

Comment:
Would it make sense to merge this if block with the ones below? seems kinda redundant to set an empty return list under 79 again.

Reply:
For clarification: under what circumstances does perses return `None` instead of raising an `InvalidMappingException`?

Reply:
@IAlibay : it needs to be a list before it gets to `preserve_chirality`. But I'd recommend an early exit for the special case of no mappings. Easiest way to do that would be for the `except` clause to set `_atom_mappings = None` (instead of `[]`) and have this be

```python
if _atom_mappings is None:
    return
```

Then you don't need the `if len(_atom_mappings)` check.

Alternate (possibly better): if `_atom_mappings` is normalized to a list, the translate process can be a comprehension, instead of a `map`:

```
mapping_dict = (x.old_to_new_atom_map for x in in _atom_mappings)
```

This will be valid for the case where `_atom_mappings == []`, and removes the need for the if statement.

Reply:
> For clarification: under what circumstances does perses return `None` instead of raising an `InvalidMappingException`?

hm... I looked in that and it is not entirely trivial. the Exception is thrown if  allow_ring_breaking set to False - leading to an unmapping. if this fails somehow, it will throw an InvalidMappingException.